Sage/carbon
# fix(radio-button): when `labelHelp` prop is provided, ensure help tooltip is displayed on focus - FE-4442

closes #4426

### Proposed behaviour

Let inner `Label` component consume input context and pass consumed values to its child components. This should fix issues such as the `RadioButton` example provided in the original issue. The corrected behaviour for `RadioButton` is demonstrated in the generated codesandbox.

Add conditional styling to `ValidationIcon`  such that, on hover, the cursor becomes a pointer if an action will be triggered on clicking the icon. This is to avoid confusion with users who would expect an action to follow if the icon is clickable.

**Remove `Label` tests that test existing state management code.**

### Current behaviour

`Label` has broken state management code, leading to regressions such as the `RadioButton` example in the original issue.

Hovering over a `ValidationIcon` part of radio controls changes the cursor to a pointer, even though there is no callback on click. The behaviour is demonstrated below:

https://user-images.githubusercontent.com/18368713/143474457-b0364279-2db4-4f0c-ab31-126d997fa677.mov





### Checklist




#### QA


### Testing instructions


I believe a test would need writing for `ValidationIcon` and `Help`, to check that the **cursor does not become a pointer when the icon is hovered over and has no `onClick` callback**. This is to avoid confusing users who would be expecting some action to occur after they click on the icon.

[The codesandbox for current behaviour can be found here.](https://codesandbox.io/s/charming-pike-9ljx5)

You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
